### DISCUSSION TITLE
Transaction pool connection limit

### DISCUSSION THREAD

**Discussion by nimish-ks** [planetscale/discussion#434] [Q&A] [answered]
Hello :wave: 

I have a cloudflare worker set up with planetscale via a serverless JavaScript driver. https://planetscale.com/docs/tutorials/planetscale-serverless-driver

I have a write heavy workload. I do INSERT on every http requests made to the worker. This works well for the most part except when I was doing a load test with k6 before going to prod. I ran into a lot of `ResourceExhasuted` issues which are causing data loss.
```
target: my-db.-.primary: vttablet: rpc error: code = ResourceExhausted desc = transaction pool connection limit exceeded
```

I was under the impression that planetscale would handle the connection pooling and load balancing for me. On the [Planetscale workflow docs page](https://planetscale.com/docs/concepts/planetscale-workflow ) it mentions 'Built-in connection pooling means you’ll never run into connection limits for your database.' 

Am I missing something? Any help or advice is highly appreciated :pray:

**Comment by mscoutermarsh** [comment, marked as answer]
Hi @nimish-ks!

That error is specifically referring to the number of concurrent transactions that are running on the database at once. It's not related to connection pooling (even though the error message is similar).

It most often occurs when there are many concurrent writes happening on a Hobby plan level database (which has more limited resources).

Could you [contact support ](https://support.planetscale.com/hc/en-us)with your account and database info? They'll be able to see specifically what is happening and give better advice.
